runtime/list.go

```diff
@@ -228,6 +228,38 @@ func listDelItem(f *Frame, o *Object, key *Object) *BaseException {
 	return l.DelItem(f, index)
 }
 
+func listDelSlice(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
+	argc := len(args)
+	if argc != 3 {
+		return nil, f.RaiseType(TypeErrorType, fmt.Sprintf("function takes exactly 2 arguments (%d given)", argc-1))
+	}
+	l := toListUnsafe(args[0])
+	if args[1].typ.slots.Index == nil || args[2].typ.slots.Index == nil {
+		return nil, f.RaiseType(TypeErrorType, "an integer is required")
+	}
+	l.mutex.Lock()
+	numListElems := len(l.elems)
+	start, raised := calcIndex(f, args[1], numListElems)
```

Comment:
Can we instead construct a slice and call DelSlice?

Reply:
@trotterdylan was thinking same, but some tests are not passing:
```
--- FAIL: TestListDelSlice (0.00s)
	list_test.go:122: TestListDelSlice([1, 2, 3, 4, 5], -1, 3) = [1, 2, 3, 4, 5], want [4, 5]
	list_test.go:122: TestListDelSlice([1, 2, 3, 4, 5], 1, -3) = [1, 3, 4, 5], want [1, 2, 3, 4, 5]
```
from CPython:
```
>>> x=[1,2,3,4,5]
>>> x.__delslice__(-1, 3)
>>> x
[4, 5]
>>> x=[1,2,3,4,5]
>>> x.__delslice__(1, -3)
>>> x
[1, 2, 3, 4, 5]
>>> 
```

Reply:
@trotterdylan ok, I will keep index alignment before initiating `Slice`, it should be enough